src/gamestate/system_state.cpp

```diff
@@ -2978,6 +2978,26 @@ void state::load_scenario_data(parsers::error_handler& err) {
 	if(gov_error)
 		return;
 
+	// Sanity checking navies & armies
+	for(auto n : world.in_navy) {
+		auto p = n.get_navy_location().get_location();
```

Comment:
while you are there, why not loop over the ships in the navy and assert that they all have a valid unit type (and for regiments below)